hub/policy/src/test/java/uk/gov/ida/hub/policy/controllogic/TransitionStoreTest.java

```diff
@@ -0,0 +1,408 @@
+        }};
+        ConcurrentMap transition = new ConcurrentHashMap() {{
+            put(new SessionId("key2"), "value2");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.of(transition));
+        assertThat(transitionStore.containsValue("value1")).isTrue();
+        assertThat(transitionStore.containsValue("value2")).isFalse();
+    }
+
+    @Test
+    public void getReturnsValueFromExisting() {
+        final SessionId key1 = new SessionId("key1");
+        ConcurrentMap existing = new ConcurrentHashMap() {{
+            put(key1, "value1");
+        }};
+        ConcurrentMap transition = new ConcurrentHashMap() {{
+            put(key1, "value2");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.of(transition));
+        assertThat(transitionStore.get(key1)).isEqualTo("value1");
+    }
+
+    @Test
+    public void putPutsValueInBothMapsAndReturnsThePreviousValueFromExisting() {
+        final SessionId key1 = new SessionId("key1");
+        ConcurrentMap existing = new ConcurrentHashMap() {{
+            put(key1, "value1");
+        }};
+        ConcurrentMap transition = new ConcurrentHashMap() {{
+            put(key1, "value2");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.of(transition));
+        assertThat(transitionStore.put(key1, "value3")).isEqualTo("value1");
+        assertThat(existing.get(key1)).isEqualTo("value3");
+        assertThat(transition.get(key1)).isEqualTo("value3");
+    }
+
+    @Test
+    public void removeRemovesFromBothQueuesAndReturnsValueFromExisting() {
+        final SessionId key1 = new SessionId("key1");
+        ConcurrentMap existing = new ConcurrentHashMap() {{
+            put(key1, "value1");
+        }};
+        ConcurrentMap transition = new ConcurrentHashMap() {{
+            put(key1, "value2");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.of(transition));
+        assertThat(transitionStore.remove(key1)).isEqualTo("value1");
+        assertThat(existing.containsKey(key1)).isFalse();
+        assertThat(transition.containsKey(key1)).isFalse();
+    }
+
+    @Test
+    public void putAllPutsAllInBothQueues() {
+        ConcurrentMap existing = new ConcurrentHashMap();
+        ConcurrentMap transition = new ConcurrentHashMap();
+        SessionId key3 = new SessionId("key3");
+        SessionId key2 = new SessionId("key2");
+        Map putAll = new HashMap() {{
+            put(key2, "value2");
+            put(key3, "value3");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.of(transition));
+        transitionStore.putAll(putAll);
+        assertThat(existing.containsKey(key2) && existing.containsKey(key3)).isTrue();
+        assertThat(existing.containsKey(key2) && existing.containsKey(key3)).isTrue();
+    }
```

Comment:
Can this line be assertThat(transition.containsKey(key2) && transition.containsKey(key3)).isTrue();?